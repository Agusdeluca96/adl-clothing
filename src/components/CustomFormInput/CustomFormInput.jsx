import React from 'react';

import './CustomFormInput.scss';

const CustomFormInput = ({ handleChange, label, ...otherCustomFormInputProps }) => (
  <div className="custom-form-input">
    <input
      className="form-input"
      onChange={handleChange}
      {...otherCustomFormInputProps}
    />
    {
      label ? (
        <label
          className={`${
            otherCustomFormInputProps.value.length ? 'shrink' : ''
            } form-input-label`}
        > {label} </label>
      ) : null
    }
  </div>
);

export default CustomFormInput;