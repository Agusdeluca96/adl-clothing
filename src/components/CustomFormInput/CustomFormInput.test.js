import React from 'react';
import { shallow } from 'enzyme';

import CustomFormInput from './CustomFormInput';

describe('CustomFormInput component', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: 'email',
      value: 'test@gmail.com',
      handleChange: mockHandleChange
    };

    wrapper = shallow(<CustomFormInput {...mockProps} />);
  });

  it('should render CustomFormInput component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange method when input changes', () => {
    wrapper.find('.form-input').simulate('change');

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it('should render form input if there is a label', () => {
    expect(wrapper.exists('.form-input-label')).toBe(true);
  });

  it('should not render form input if there is no label', () => {
    const mockNewProps = {
      label: '',
      value: 'test@gmail.com',
      handleChange: mockHandleChange
    };

    const newWrapper = shallow(<CustomFormInput {...mockNewProps} />);

    expect(newWrapper.exists('.form-input-label')).toBe(false);
  });
});
