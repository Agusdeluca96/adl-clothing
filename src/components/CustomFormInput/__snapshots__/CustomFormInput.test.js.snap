// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomFormInput component should render CustomFormInput component 1`] = `ShallowWrapper {}`;
